### PR TITLE
Bulls Eye Skill

### PR DESCRIPTION
**Pull request by ImMuscleMan**
## Describe your changes
Add a Bulls Eye skill to ranger as a sword ability
https://github.com/Mykindos/BetterPvP/assets/95445240/1754b8cf-aee1-45a2-86db-f41ea97d54c2

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested my changes.

### CODE REVIEW

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/data/BullsEyeData.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
+package me.mykindos.betterpvp.champions.champions.skills.skills.ranger.data;
+
+import me.mykindos.betterpvp.champions.champions.skills.data.ChargeData;
+import org.bukkit.*;
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/data/BullsEyeData.java:4]: avoid using wildcard imports, you can change this in intellij

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/sword/Bullseye.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+import me.mykindos.betterpvp.champions.champions.skills.types.InteractSkill;
+import me.mykindos.betterpvp.core.client.gamer.Gamer;
+import me.mykindos.betterpvp.core.combat.events.CustomDamageEvent;
+import me.mykindos.betterpvp.core.components.champions.Role;
+import me.mykindos.betterpvp.core.components.champions.SkillType;
+import me.mykindos.betterpvp.core.framework.updater.UpdateEvent;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import org.bukkit.*;
+import org.bukkit.entity.*;
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/sword/Bullseye.java:22]: more wildcards

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/data/BullsEyeData.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            return Color.fromRGB(235, 235, 40);
+        } else if (targetFocusedAmount <= 0.72) {
+            return Color.fromRGB(205, 240, 40);
+        } else if (targetFocusedAmount <= 0.81) {
+            return Color.fromRGB(170, 240, 40);
+        } else if (targetFocusedAmount <= 0.9) {
+            return Color.fromRGB(110, 240, 40);
+        } else if (targetFocusedAmount <= 1.0) {
+            return Color.fromRGB(70, 240, 40);
+        }
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/data/BullsEyeData.java:104]: Theres probably a cleaner way to do this

Perhaps multiply R (250) by 1-targetFocusedAmount, and multiply G (50) by 1+targetFocusedAmount

**Review by Mykindos — CHANGES_REQUESTED** [review]

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/data/BullsEyeData.java (hunk 3 above)
Comment (ImMuscleMan) [reply champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/data/BullsEyeData.java:104]: Yeah, I tried a few ways but most seemed to go into dark and dull colours which didn't look too good, I'll try some other methods and see what I can come up with

**Review by Mykindos — CHANGES_REQUESTED** [review]

**Review by Mykindos — APPROVED** [review]

**Pull request merged**